### PR TITLE
fix(ccp): manage state in the utility thread

### PR DESCRIPTION
**Pull request by monoid**
It avoids race conditions when file system is accessed from different threads.

It adds new optional parameter `control_queue_size` to `[rpc-endpoint]`.  Generally, you should not tweak it.

### CODE REVIEW

File: ccp/Cargo.toml [hunk 1]
```diff
@@ -44,6 +44,7 @@ thiserror.workspace = true
 axum = "0.7.4"
 prometheus-client = "0.22.1"
 tokio-util = "0.7.10"
```
Comment (raftedproc) [inline ccp/Cargo.toml:46]: Should be grouped with the other provided-by-workspace dependencies IMHO.

File: ccp/src/cu/proving_thread/tests.rs [hunk 2]
```diff
@@ -43,10 +44,15 @@ impl ThreadInitIngredients {
     pub(self) async fn create(core_id: LogicalCoreId, epoch: EpochParameters, cu_id: CUID) -> Self {
         let flags = RandomXFlags::recommended_full_mem();
 
-        let (inlet, outlet) = mpsc::channel(1);
+        let (inlet, outlet) = mpsc::channel(3);
 
+        let thread_cfg = ProvingThreadConfig {
+            hashes_per_round: 1024,
```
Comment (raftedproc) [inline ccp/src/cu/proving_thread/tests.rs:50]: Don't we have a constant for hashes per round?

File: ccp/src/errors.rs [hunk 3]
```diff
@@ -40,6 +40,9 @@ pub enum CCProverError {
 
     #[error(transparent)]
     IOError(#[from] tokio::io::Error),
+
+    #[error("utility thread closed channel")]
```
Comment (raftedproc) [inline ccp/src/errors.rs:44]: The comment [here](https://github.com/fluencelabs/capacity-commitment-prover/pull/127/files#diff-f9b9eeefbbfbeea42d236f5ab6677f16f0b54a83479eb853ae3af09b1d46c053R338) tells the full queue causes the error. I think it is worth to sync the comment and the error msg.

**Review by raftedproc — APPROVED** [review]
uACK except minor comments-related notes.

File: ccp/Cargo.toml (hunk 1 above)
Comment (monoid) [reply ccp/Cargo.toml]: Indeed, this crate is used in more that one crate.

File: ccp/src/errors.rs (hunk 3 above)
Comment (monoid) [reply ccp/src/errors.rs]: Indeed. Refactored utility errors.

**Pull request merged**